### ISSUE TITLE
Specifying an analyzer for the _all field breaks mapping updates

### ISSUE THREAD

**Issue by lperry** [elastic/elasticsearch#12407]
Specifying an analyzer for the _all field breaks mapping updates.

```
curl -X DELETE http://localhost:29200/twitter
# {"acknowledged":true}

curl -X PUT http://localhost:29200/twitter -d '
{
  "mappings": {
    "tweet": {
      "_all": {
        "analyzer": "snowball"
      },
      "properties": {
        "tweet": {
          "type": "string",
          "analyzer": "english"
        }
      }
    }
  }
}
# {"acknowledged":true}

curl -X PUT http://localhost:29200/twitter/_mapping/tweet -d '
{
  "properties": {
    "tag": {
      "type": "string",
      "index": "not_analyzed"
    }
  }
}
'
# {"error":"MergeMappingException[Merge failed with failures {[mapper [_all] has different index_analyzer]}]","status":400}
```

If no analyzer is specified for the _all field when creating the index, the mapping update call succeeds.

I've tested the above against Elasticsearch 1.7.0.

Thanks!

Perry

**Comment by lperry** [comment]
It looks like specifying a new field with the same analyzer as the _all field also returns an error:
```
curl -X PUT http://localhost:29200/twitter/_mapping/tweet -d '
{
  "properties": {
    "tag": {
      "type": "string",
      "analyzer": "snowball"
    }
  }
}
'
# {"error":"MergeMappingException[Merge failed with failures {[mapper [_all] has different index_analyzer]}]","status":400}
```

**Comment by clintongormley** [comment]
Hi @lperry 

You can't change an analyzer after the field has been created.  It would invalidate your existing data.

**Comment by lperry** [comment]
Hi @clintongormley,

I'm not trying to change an existing field's analyzer. I'm trying to add a new field.

The provided curl commands first create an index with an analyzer specified for the _all field and a field tweet, and then tries to add a new tag field.

Thanks,

Perry

**Comment by clintongormley** [comment]
Apologies @lperry - I misread the issue.  You are correct.  This has already been fixed in 2.0.

For now you can work around it by respecifying the `_all` mapping:

    PUT /twitter/_mapping/tweet
    {
      "_all": {
        "analyzer": "snowball"
      },
      "properties": {
        "tag": {
          "type": "string",
          "index": "not_analyzed"
        }
      }
    }

**Comment by lperry** [comment]
Thanks for the workaround and fix!
